app/ide-desktop/lib/client/src/ipc.ts

```diff
@@ -12,4 +12,40 @@ export const channel = {
     profilesLoaded: 'profiles-loaded',
     saveProfile: 'save-profile',
     quit: 'quit-ide',
+    /**
+     * Channel for requesting that a URL by opened by the system browser.
+     *
+     * Events on this channel are emitted by our `enso-cloud-authentication` package. These events
+     * are intended to send the user to the system browser to complete sensitive flows such as
+     * signup and signin. These flows should not be done in the app as the user cannot be expected
+     * to trust the app with their credentials.
+     */
+    openExternalUrl: 'open-external-url',
+    /**
+     * Channel for the `enso-cloud-authentication` module to **set** (not **use**!) a callback.
+     *
+     * The callback is the complement of the `openExternalUrl` channel above, and is intended for
+     * opening URLs in the app, rather than in the system browser. The callback is called by the
+     * Electron app when a user is sent back to the app from their system browser via a deep link.
+     * The URL requested in the deep link is sent to the callback. The `enso-cloud-authentication`
+     * module then reads this URL and performs the necessary redirects.
+     *
+     * This flow is complex. The simpler answer would be to simply let the Electron app handle the
+     * deep link without us catching it and sending it to the callback. But this won't work because
+     * we have strict requirements about not redirecting the user to any URL. All our in-app
+     * redirects are emulated via browser history, since the user never leaves our "index" page.
+     * This is done for security reasons (to avoid the user getting redirected to a malicious URL)
+     * and for performance reasons (reloading the page would cause a full re-render of our
+     * authentication and dashboard UIs).
```

Comment:
TBH I did not nunderstand much from this comment. Just a few points:
1. `setOpenAuthenticationUrlCallback` - if this is intented to open urls in the app rather than in system browser, then I dont understand the name of the var.
2. `This flow is complex.` - please describe it step by step, preferably in points, in such a way that it is understandable for the reader (right now its not) and in such a way that the description does not need to contain infromation that it is complex.
3. `The simpler answer would be ...` - answer to what?
4. `The simpler answer would be to simply let the Electron app handle the deep link without us catching it and sending it to the callback. But this won't work because we have strict requirements about not redirecting the user to any URL`. So if it will not work, it would not be an answer/solution, right?
5. `All our in-app redirects are emulated via browser history` - what does it mean? This is the first time I hear about emulating something with browser history/ Where are docs about it and explanation of this design?
6. `This is done for security reasons ... and for performance reasons` - I think the most important point is missed here. If you reload the app, you lose your work. If we would allow users to log out / log in during using the app, then reloading would just remove all. the current state.